### PR TITLE
feat: lottery opt in, history (release)

### PR DESCRIPTION
**Pull request by emilyjablonski**
Releases the below commits from core

### CODE REVIEW

File: api/src/services/lottery.service.ts [hunk 1]
```diff
@@ -224,4 +289,702 @@
...
+      }-${new Date().getTime()}.zip`,
+    );
+
+    await this.createLotterySheets(workbook, {
+      ...queryParams,
+    });
+
+    await workbook.xlsx.writeFile(filename);
+
+    const readStream = createReadStream(filename);
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply api/src/services/lottery.service.ts:329]: We're verifying that the user input here is a UUID

File: api/src/services/lottery.service.ts [hunk 2]
```diff
@@ -224,4 +289,702 @@
...
+
+    return new Promise((resolve) => {
+      // Create a writable stream to the zip file
+      const output = fs.createWriteStream(zipFilePath);
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply api/src/services/lottery.service.ts:333]: We're verifying that the user input here is a UUID

File: api/src/services/lottery.service.ts [hunk 3]
```diff
@@ -224,4 +289,702 @@
...
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      output.on('close', () => {
+        const zipFile = createReadStream(zipFilePath);
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply api/src/services/lottery.service.ts:338]: We're verifying that the user input here is a UUID

**Review by ColinBuyck — APPROVED** [review]
LGTM 🪀

**Pull request merged**